### PR TITLE
libsigsegv: bump to 2.12, add debuginfo.

### PR DESCRIPTION
**Pull request by fbrosson**

### CODE REVIEW

**Comment by fbrosson** [comment]
Thils will close #2130.
@bhaible, did I put the right years in every line of [`COPYRIGHT`](https://github.com/fbrosson/haikuports/blob/libsigsegv/dev-libs/libsigsegv/libsigsegv-2.12.recipe#L16)? Thanks!

File: dev-libs/libsigsegv/libsigsegv-2.12.recipe [hunk 1]
```diff
@@ -13,30 +13,32 @@ This library supports three sets of functions, all defined in <sigsegv.h>:
...
+	lib:libsigsegv$secondaryArchSuffix = $libVersionCompat
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
 	"
 
 PROVIDES_devel="
 	libsigsegv${secondaryArchSuffix}_devel = $portVersion
-	devel:libsigsegv$secondaryArchSuffix = $portVersion
+	devel:libsigsegv$secondaryArchSuffix = $libVersion
```
Comment (fbrosson) [inline dev-libs/libsigsegv/libsigsegv-2.12.recipe]: @alaviss, do you know if I should use `$libVersionCompat` here, instead of `$libVersion`? Thanks!
(I need to learn how to find the answer by myself...)

**Comment by bhaible** [comment]
> @bhaible, did I put the right years in every line of [`COPYRIGHT`]

Yes, this is correct.

File: dev-libs/libsigsegv/libsigsegv-2.12.recipe (hunk 1 above)
Comment (alaviss) [reply dev-libs/libsigsegv/libsigsegv-2.12.recipe]: So here's how it works. A shared library typically provides an ABI/API compatibility version, which means software linked with the old version is guaranteed to work with the new release of the same compatibility version (not always the case). This means that shared library could be swapped to a new version that fixes security/minor issues and program linked to it would still work. To achieve this, the API must also supports the same thing that the older versions (within compatibility version) does. This means it's safe to provides a compat version for the devel package as well.

TL;DR: You should use `$libVersionCompat`, as ports that use the version 2 of the API could do this

```bash
BUILD_REQUIRES="
	...
	devel:libsigsegv$secondaryArchSuffix >= 2
	...
	"
```

instead of

```bash
BUILD_REQUIRES="
	...
	devel:libsigsegv$secondaryArchSuffix >= 2.0.5
	...
	"
```

Comment (fbrosson) [reply dev-libs/libsigsegv/libsigsegv-2.12.recipe]: Thanks @alaviss.
Actually, I already knew which is the correct setting (between `$libVersionCompat` and `$libVersion`) for the `lib:libsigsegv` in `PROVIDES`, as well as for any lib in any package. The answer depends on whether or not the "make install" installs the symbolic link that "acts" as a "compat".
 ~What I don't know is wether or not the setting for `PROVIDES_devel` should always be the same as the setting for `PROVIDES`.~

OK, I think I got it now.

I have updated the recipe to use `$libVersionCompat` (instead of `$libVersion`) in `PROVIDES_devel`.
I guess it is OK now (but I'll wait for a confirmation before I merge :)
